lte/cloud/go/services/subscriberdb/storage/sub_proto.go

```diff
@@ -0,0 +1,335 @@
+)
+
+var (
+	once       sync.Once
+	sqlDialect string
+)
+
+type SubProtoStore struct {
+	db      *sql.DB
+	builder sqorc.StatementBuilder
+}
+
+func NewSubProtoStore(db *sql.DB, builder sqorc.StatementBuilder) *SubProtoStore {
+	return &SubProtoStore{db: db, builder: builder}
+}
+
+// getSqlDialect looks up the sql dialect of the current environment once and stores it
+// in memory cache for use.
+func getSqlDialect() string {
+	once.Do(func() {
+		dialect, envFound := os.LookupEnv("SQL_DIALECT")
+		// Default to postgresql
+		if !envFound {
+			sqlDialect = sqorc.PostgresDialect
+		} else {
+			sqlDialect = dialect
+		}
+	})
+	return sqlDialect
+}
+func (l *SubProtoStore) Initialize() error {
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		_, err := l.builder.CreateTable(subProtoTableName).
+			IfNotExists().
+			Column(subProtoNidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(subProtoSidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(subProtoProtoCol).Type(sqorc.ColumnTypeBytes).NotNull().EndColumn().
+			PrimaryKey(subProtoNidCol, subProtoSidCol).
+			RunWith(tx).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrap(err, "initialize sub proto store table")
+		}
+		_, err = l.builder.CreateTable(subProtoTmpTableName).
+			IfNotExists().
+			Column(subProtoNidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(subProtoSidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(subProtoProtoCol).Type(sqorc.ColumnTypeBytes).NotNull().EndColumn().
+			PrimaryKey(subProtoNidCol, subProtoSidCol).
+			RunWith(tx).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrap(err, "initialize sub proto store tmp table")
+		}
+		return nil, nil
+	}
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+// InsertManyByNetwork serializes and inserts a batch of subscriber protos into the temporary table.
+func (l *SubProtoStore) InsertManyByNetwork(network string, subProtos []*lte_protos.SubscriberData) error {
+	if len(subProtos) == 0 {
+		return nil
+	}
+	insertQuery := l.builder.
+		Insert(subProtoTmpTableName).
+		Columns(subProtoNidCol, subProtoSidCol, subProtoProtoCol)
+	errs := &multierror.Error{}
+	for _, subProto := range subProtos {
+		marshaled, err := proto.Marshal(subProto)
+		if err != nil {
+			multierror.Append(errs, errors.Wrapf(err, "serialize subproto of ID %+v", lte_protos.SidString(subProto.Sid)))
+			continue
+		}
+		insertQuery = insertQuery.Values(network, lte_protos.SidString(subProto.Sid), marshaled)
+	}
+	if errs.ErrorOrNil() != nil {
+		return errs
+	}
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		_, err := insertQuery.RunWith(tx).Exec()
+		return nil, errors.Wrapf(err, "insert sub protos into store for network %+v", network)
+	}
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+// CommitUpdateByNetwork cleans up and re-populates the subscriber proto store table with data from the temporary
```

Comment:
From other comment: docstrings should tell "what" the fn is doing (the abstract conceptual affordances the caller cares about) instead of "how" the fn is doing it (these are implementation details, and belong either inside the fn or in a comment on the object defn